frontend/src/ui-components/AcmTable/AcmManageColumn.tsx

```diff
@@ -0,0 +1,213 @@
+    isModalOpen,
+    toggleModal,
+    allCols,
+    selectedColIds,
+    setSelectedColIds,
+    colOrderIds,
+    setColOrderIds,
+    defaultColIds,
+  } = props
+  const [items, setItems] = useState<IAcmTableColumn<T>[]>(sortByList(colOrderIds, allCols))
+  const [localSelectedIds, setlocalSelectedIds] = useState<string[]>(selectedColIds)
+
+  const onDrop = (source: any, dest?: any) => {
+    if (dest) {
+      const newItems = reorder<T>(items, source.index, dest.index)
+      setItems(newItems)
+      return true
+    } else {
+      return false
+    }
+  }
+
+  const handleChange = (checked: boolean, event: FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.name
+    if (checked) {
+      setlocalSelectedIds([...localSelectedIds, value])
+    } else {
+      setlocalSelectedIds(localSelectedIds.filter((id) => id !== value))
+    }
+  }
+
+  const onSave = () => {
+    setSelectedColIds(localSelectedIds)
+    setColOrderIds(items.map((col) => col.id!))
+    toggleModal()
+  }
+
+  const restoreDefault = () => {
+    setlocalSelectedIds(defaultColIds)
+    const sortedItems = [...items].sort((a, b) => {
+      return a.order != null && b.order != null ? a.order - b.order : -1
+    })
+    setItems(sortedItems)
+  }
+
+  const onClose = () => {
+    toggleModal()
+    setlocalSelectedIds(selectedColIds)
+    // sort listed column by saved column order
+    setItems(sortByList(colOrderIds, items))
+  }
+
+  return (
+    <Modal
+      title={t('Manage columns')}
+      isOpen={isModalOpen}
+      variant="small"
+      description={
+        <TextContent>
+          <Text>{t('Selected columns will appear in the table. Drag and drop the columns to reorder them.')}</Text>
+        </TextContent>
+      }
+      onClose={onClose}
+      actions={[
+        <Button key="save" variant="primary" onClick={onSave}>
+          {t('Save')}
+        </Button>,
+        <Button key="cancel" variant="secondary" onClick={onClose}>
+          {t('Cancel')}
+        </Button>,
+        <Button key="restore" variant="link" onClick={restoreDefault}>
+          {t('Restore defaults')}
+        </Button>,
+      ]}
+    >
+      <DragDrop onDrop={onDrop}>
+        <Droppable hasNoWrapper>
+          <DataList aria-label="Table column management" id="table-column-management" isCompact>
+            {items.map((policy) => (
+              <Draggable key={policy.id} hasNoWrapper>
+                <DataListItem aria-labelledby={`table-column-management-${policy.id}`} id={policy.id}>
+                  <DataListItemRow>
+                    <DataListControl>
+                      <DataListDragButton
+                        aria-label="Reorder"
+                        aria-labelledby={`table-column-management-${policy.id}`}
+                        aria-describedby="Press space or enter to begin dragging, and use the arrow keys to navigate up or down. Press enter to confirm the drag, or any other key to cancel the drag operation."
```

Comment:
I would suggest we translate this to support accessibility in other languages.

Reply:
Good catch Thanks!